iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -388,21 +388,13 @@ public static PartitionData toPartitionData(StructLike sourceKey, Types.StructTy
       Types.StructType targetKeyType) {
     PartitionData data = new PartitionData(targetKeyType);
     for (int i = 0; i < targetKeyType.fields().size(); i++) {
```

Comment:
could we replace loop with foreach. it seems we don't actually need index, but a `field`